playbooks/ceph.yml

```diff
@@ -366,30 +359,99 @@
 
-    # cifmw_cephadm_vip is the VIP reserved in the RGW network
-    - name: Set VIP var as empty string
-      ansible.builtin.set_fact:
-        cifmw_cephadm_vip: ""
+        - name: Set VIP var as empty string
+          ansible.builtin.set_fact:
+            cifmw_cephadm_vip: ""
+          when:
+            - cifmw_cephadm_rgw_vip is undefined
 
-    - name: Process VIP
-      ansible.builtin.include_role:
-        name: cifmw_cephadm
-        tasks_from: check_vip
-      loop: "{{ range(1, (ips | length) + 1) | list }}"
+        - name: Get RGW VIP
+          ansible.builtin.include_role:
+            name: cifmw_cephadm
+            tasks_from: check_vip
+          loop: "{{ range(1, (ips | length) + 1) | list }}"
+          vars:
+            cifmw_cephadm_vip_network: "{{ cifmw_cephadm_rgw_network | default(storage_network_range, true) | default(ssh_network_range, true) }}"
+          when:
+            - cifmw_cephadm_rgw_vip is undefined
+
+        - name: Set RGW VIP
+          ansible.builtin.set_fact:
+            cifmw_cephadm_rgw_vip: "{{ cifmw_cephadm_vip }}/{{ cidr }}"
+          when:
+            - cifmw_cephadm_rgw_vip is undefined
+
+    - name: Set NFS Network Properties
+      when:
+        - cifmw_ceph_daemons_layout.ceph_nfs_enabled | default(false) | bool
+      block:
+        - name: Set NFS network range to storage network only if it was not provided
+          ansible.builtin.set_fact:
+            cifmw_cephadm_nfs_network: "{{ cifmw_cephadm_storage_network }}"
+          when:
+            - cifmw_cephadm_nfs_network is not defined or
+              cifmw_cephadm_nfs_network | length == 0
+
+        - name: Assert if any EDPM nodes n/w interface is missing in NFS network
+          ansible.builtin.assert:
+            that:
+              - hostvars[item][all_addresses] | ansible.utils.ipaddr(cifmw_cephadm_nfs_network) | length > 0
+            fail_msg: "node {{ item }} doesn't have any interface connected to network {{ cifmw_cephadm_nfs_network }}"
+          loop: "{{ _target_hosts }}"
+          when:
+            - cifmw_cephadm_nfs_network != cifmw_cephadm_storage_network
+
+        - name: Get already assigned NFS IP addresses
+          ansible.builtin.set_fact:
+            ips: "{{ ips | default([]) + [ hostvars[item][all_addresses] | ansible.utils.ipaddr(cifmw_cephadm_nfs_network) | first ] }}"
+          loop: "{{ _target_hosts }}"
+
+        - name: Set VIP var as empty string
+          ansible.builtin.set_fact:
+            cifmw_cephadm_vip: ""
+          when:
+            - cifmw_cephadm_nfs_vip is undefined
+
+        - name: Get NFS VIP
+          ansible.builtin.include_role:
+            name: cifmw_cephadm
+            tasks_from: check_vip
+          loop: "{{ range(1, (ips | length) + 1) | list }}"
+          vars:
+            cifmw_cephadm_vip_network: "{{ cifmw_cephadm_nfs_network | default(storage_network_range, true) | default(ssh_network_range, true) }}"
```

Comment:
I assume that by default we pass `cifmw_cephadm_vip_network` in our jobs to point to the NFS network instead of storage network, right?

Reply:
yes, something like this.

```
cifmw_cephadm_nfs_network: 172.21.0.0/24

# override variables in hooks/playbooks/manila_create_default_resources.yml
manila_provider_network_name: nfs
manila_provider_network_vlan: 24
manila_provider_network_start: 172.21.0.150
manila_provider_network_end: 172.21.0.200
manila_provider_network_range: 172.21.0.0/24

# Attach test pod to nfs network (Kind: Tempest spec/networkAttachments: [nfs])
cifmw_test_operator_tempest_network_attachments:
  - ctlplane
  - nfs

```